bin/crew

```diff
@@ -1847,6 +1857,111 @@ def print_deps_tree(args)
   puts treeView
 end
 
+def upload(args)
+  abort "\nGITLAB_TOKEN environment variable not set.\n".lightred if ENV['GITLAB_TOKEN'].nil?
+
+  packages = ''
+  gitlab_token = ENV['GITLAB_TOKEN']
+
+  if ARGV.length > 0
+    ARGV.each do |package|
+      unless package == 'upload'
+        pkgfile = "#{CREW_LOCAL_REPO_ROOT}/packages/#{package}.rb"
+        if File.exist?(pkgfile)
+          packages += " #{package}"
+        else
+          puts "Package #{pkgfile} not found.".orange
+        end
+      end
+    end
+  else
+    %w[.tar.xz .tpxz .tar.zst].each do |ext|
+      packages += `find #{CREW_LOCAL_REPO_ROOT}/release/*/*#{ext} -exec basename -s #{ext} {} + | cut -d- -f1 | sort | uniq | xargs`
+      packages += ' '
+    end
+    abort 'No package binaries found.'.lightred if packages.empty?
+  end
+
+  base_url = 'https://gitlab.com/api/v4/projects/26210301/packages/generic'
+
+  packages.strip!
+
+  [packages].each do |package|
+    %w[x86_64 i686 armv7l].each do |arch|
+      puts "Package: #{package}, Arch: #{arch}".yellow
+      pkgfile = "#{CREW_LOCAL_REPO_ROOT}/packages/#{package}.rb"
+      new_tarfile = Dir["#{CREW_LOCAL_REPO_ROOT}/release/#{arch}/#{package}-*-chromeos-#{arch}.{tar.xz,tpxz,tar.zst}"].max_by { |f| File.mtime(f) }
+      if new_tarfile.nil?
+        puts "#{CREW_LOCAL_REPO_ROOT}/release/#{arch}/#{package}-#-chromeos-#{arch}.(tar.xz|tpxz|tar.zst) not found.\n".lightred
+        next
+      end
+      puts 'Generating sha256sum ...'
+      new_sha256 = `sha256sum #{new_tarfile} | cut -d' ' -f1`.chomp
```

Comment:
I believe there is a built in ruby function for this, which we should be using instead of system commands.

Reply:
I wasn't able to find a command.  Feel free to suggest something else.

Reply:
We use Digest::SHA256.hexdigest(File.read('<path>')) in a number of places around the codebase, so probably that.

Reply:
That won't work.  We need to look for the hash of a specific architecture.